engine/src/control.cpp

```diff
@@ -454,6 +454,60 @@ void MCControl::deselect()
 	}
 }
 
+void MCControl::removefromobjecttree(bool p_is_child)
+{
+    // IM-2012-05-16 [[ BZ 10212 ]] deleting the dragtarget control in response
+    // to a 'dragdrop' message would leave these globals pointing to the deleted
+    // object, leading to an infinite loop if the target was a field
+    if (MCdragdest == this)
+    {
+        MCdragdest = nil;
+        MCdropfield = nil;
+    }
+    
+    if (MCdragsource == this)
+        MCdragsource = nil;
+	
+    // IM-2016-10-05: [[ Bug 17008 ]] Dirty selection handles when object deleted
+    if (getselected())
+        getcard()->dirtyselection(rect);
+    
+	// Remove from the id cache before detaching from parent
+	MCObject::removefromobjecttree(p_is_child);
+    
+    switch (parent->gettype())
+    {
+        case CT_CARD:
+        {
+            if (!p_is_child)
+            {
+                parent.GetAs<MCCard>()->removecontrol(this, False, True);
+            }
+            getstack()->removecontrol(this);
+            break;
+        }
+        case CT_GROUP:
+        {
+            if (!p_is_child)
+            {
+                parent.GetAs<MCGroup>()->removecontrol(this, True);
+            }
+            break;
+        }
+        default:
+        {
+            parent.GetAs<MCStack>()->removecontrol(this);
+        }
```

Comment:
I feel like this should be:

```C++
case CT_STACK:
    parent.GetAs<MCStack>()->removecontrol(this);
    break;
default:
    MCUnreachable();
```

This is because you're switching on the parents object type, and it makes sense to have a stack case for stacks. :)